pkg/cmd/auth/setupgit/setupgit.go

```diff
@@ -74,12 +78,28 @@ func setupGitRun(opts *SetupGitOptions) error {
 		return err
 	}
 	authCfg := cfg.Authentication()
-
 	hostnames := authCfg.Hosts()
 
 	stderr := opts.IO.ErrOut
 	cs := opts.IO.ColorScheme()
 
+	// If a hostname was provided, we'll set up just that one
+	if opts.Hostname != "" {
```

Comment:
I noticed in the existing code that `--hostname` without `--force` would return with different errors depending on whether there were known hosts or not, and that seemed like a bad UX.

To sort this out, it made more sense to just branch early based on whether a hostname had been provided or not. This also has a symmetry with the flags because we can ignore `--force` entirely after the `--hostname` block.